chia/daemon/server.py

```diff
@@ -231,74 +230,84 @@ async def incoming_connection(self, request):
                     decoded = json.loads(msg.data)
                     if "data" not in decoded:
                         decoded["data"] = {}
-                    response, sockets_to_use = await self.handle_message(ws, decoded)
+                    response, connections = await self.handle_message(ws, decoded)
+                    service_name = decoded["destination"]
                 except Exception as e:
                     tb = traceback.format_exc()
                     self.log.error(f"Error while handling message: {tb}")
                     error = {"success": False, "error": f"{e}"}
                     response = format_response(decoded, error)
-                    sockets_to_use = []
-                if len(sockets_to_use) > 0:
-                    for socket in sockets_to_use:
-                        try:
-                            await socket.send_str(response)
-                        except Exception as e:
-                            tb = traceback.format_exc()
-                            self.log.error(f"Unexpected exception trying to send to websocket: {e} {tb}")
-                            self.remove_connection(socket)
-                            await socket.close()
+                    service_name = "Unknown"
+                    connections = {ws}  # send error back to the sender
+
+                for connection in connections:
+                    if connection.closed:
+                        # strange, connection is closed, but we still have it in the list
+                        self.remove_connection(connection)
+                        continue
+                    try:
+                        await connection.send_str(response)
```

Comment:
There might still be a race where the connection has closed by now.  Maybe there's a distinct exception for that that we could catch here to respond to specifically instead of pre-checking?

Reply:
It probably throws ConnectionReset - so I was just hoping to avoid the traceback in the log for this case. Although I could write better logging in the ConnectionReset case, since that is a rather typical networking error

Reply:
I removed it